javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -0,0 +1,164 @@
+package java.util
+
+import java.io.Serializable
+
+// Added extra private constructors to handle all of the overloads.
+// To preserve method signatures, we cannot take ClassTag via implicit parameters.
+// We use an Array[AnyRef] for underlying storage and box/unbox AnyVals where needed as the JDK class would do.
+/**
+ * @param inner The underlying array
+ * @param _size The effective size of the underlying array. a.k.a. end index exclusive
+ */
```

Comment:
We don't do doc comments for the java library code. The only docs are the reference docs from the JDK.